File: spring-best/src/main/java/com/best/springbest/model/OrderDetail.java
package com.best.springbest.model;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "orderdetails")
public class OrderDetail {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "order_id",nullable = false)
    private Order order;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "warehouse_voucher_id",nullable = false)
    private WarehouseVoucher warehouseVoucher;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "product_detail_id",nullable = false)
    private ProductDetail productDetail;

    private  int quantity;

    private double price;
}

File: BEST-App-main/lib/model/order_detail.dart

class OrderDetail {
  int id;
  int orderId;
  int warehouseVoucherId;
  int productDetailId;
  int quantity;
  double price;

  OrderDetail(this.id, this.orderId, this.warehouseVoucherId, this.productDetailId, this.quantity, this.price);

  static Map<String, dynamic> toJson(OrderDetail orderDetail) {
    return {
      'id': orderDetail.id,
      'order_id': orderDetail.orderId,
      'warehouse_voucher_id': orderDetail.warehouseVoucherId,
      'product_detail_id': orderDetail.productDetailId,
      'quantity': orderDetail.quantity,
      'price': orderDetail.price,
    };
  }
}
